### PR TITLE
Add typings for passport-openidconnect@0.1.1

### PR DESCRIPTION
**Pull request by elmaxe**
Package codeowner: @jaredhanson

Looked at [@types/passport-oauth2](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/passport-oauth2) to make this definition similar

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by elmaxe** [comment]
Looked a lot at @pasieronen's @types/passport-oauth2

File: types/passport-openidconnect/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
+// Type definitions for passport-openidconnect 0.1
+// Project: https://github.com/jaredhanson/passport-openidconnect#readme
+// Definitions by: Axel Elmarsson <https://github.com/elmaxe>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
```
Comment (peterblazejewicz) [inline types/passport-openidconnect/index.d.ts:5]: hm, this sounds like the past decade only, so:
```suggestion
```

File: types/passport-openidconnect/tslint.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+                "errors": [
+                    [
+                        "NeedsExportEquals",
+                        false
+                    ]
+                ]
+            }
+        ]
+    }
+}
```
Comment (peterblazejewicz) [inline types/passport-openidconnect/tslint.json:17]: why do you need all of those? Exclusions requires special cases, sorry!

File: types/passport-openidconnect/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        constructor(message: string | undefined, code: string, uri?: string, status?: number);
+        code: string;
+        uri?: string | undefined;
+        status: number;
+    }
+
+    class InternalOAuthError extends Error {
+        constructor(message: string, err: any);
+        oauthError: any;
+    }
```
Comment (peterblazejewicz) [inline types/passport-openidconnect/index.d.ts:118]: so reall, does it work for you?
```js
const { InternalOAuthError } = require('passport-openidconnect');
```
please do not export symbols that do not exist on native module

**Review by Woodpile37 — APPROVED** [review]

**Comment by rbuckton** [comment]
@elmaxe, several reviewers have left feedback. Can you follow up?

**Comment by elmaxe** [comment]
> @elmaxe, several reviewers have left feedback. Can you follow up?

Yes I will when I have time!

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Please address feedback

**Pull request closed without merging**